tests/Unit/OutputFormatterTest.php

```diff
@@ -297,4 +299,257 @@ public function implodeWithIncreaseLevelTrueIncreasesIndentationLevelForRenderin
+
+        self::assertStringContainsString('/*' . $commentText . '*/', $result);
+    }
+
+    /**
+     * @test
+     */
+    public function commentsWithCommentableWithOneCommentPutsSpaceAfterBlocksAfterRenderedComment(): void
+    {
+        $this->outputFormat->setRenderComments(true);
+        $this->outputFormat->setSpaceBetweenBlocks(' between-space ');
+        $afterSpace = ' after-space ';
+        $this->outputFormat->setSpaceAfterBlocks($afterSpace);
+
+        $commentText = 'This is a comment.';
+        $comment = new Comment($commentText);
+        $commentable = $this->createMock(Commentable::class);
+        $commentable->method('getComments')->willReturn([$comment]);
+
+        $result = $this->subject->comments($commentable);
+
+        self::assertSame('/*' . $commentText . '*/' . $afterSpace, $result);
+    }
+
+    /**
+     * @test
+     */
+    public function commentsWithCommentableWithTwoCommentsPutsSpaceAfterBlocksAfterLastRenderedComment(): void
+    {
+        $this->outputFormat->setRenderComments(true);
+        $this->outputFormat->setSpaceBetweenBlocks(' between-space ');
+        $afterSpace = ' after-space ';
+        $this->outputFormat->setSpaceAfterBlocks($afterSpace);
+
+        $commentText1 = 'This is a comment 1.';
+        $comment1 = new Comment($commentText1);
+        $commentText2 = 'This is a comment 2.';
+        $comment2 = new Comment($commentText2);
+        $commentable = $this->createMock(Commentable::class);
+        $commentable->method('getComments')->willReturn([$comment1, $comment2]);
+
+        $result = $this->subject->comments($commentable);
+
+        self::assertStringContainsString('/*' . $commentText2 . '*/' . $afterSpace, $result);
+    }
+
+    /**
+     * @test
+     */
+    public function commentsWithCommentableWithTwoCommentsSeparatesCommentsBySpaceBetweenBlocks(): void
+    {
+        $this->outputFormat->setRenderComments(true);
+        $betweenSpace = ' between-space ';
+        $this->outputFormat->setSpaceBetweenBlocks($betweenSpace);
+        $this->outputFormat->setSpaceAfterBlocks(' after-space ');
+
+        $commentText1 = 'This is a comment 1.';
+        $comment1 = new Comment($commentText1);
+        $commentText2 = 'This is a comment 2.';
+        $comment2 = new Comment($commentText2);
+        $commentable = $this->createMock(Commentable::class);
+        $commentable->method('getComments')->willReturn([$comment1, $comment2]);
+
+        $result = $this->subject->comments($commentable);
+
+        $expected = '/*' . $commentText1 . '*/' . $betweenSpace . '/*' . $commentText2 . '*/';
+        self::assertStringContainsString($expected, $result);
+    }
+
+    /**
+     * @test
+     */
+    public function commentsWithCommentableWithMoreThanTwoCommentsPutsSpaceAfterBlocksAfterLastRenderedComment(): void
+    {
+        $this->outputFormat->setRenderComments(true);
+        $this->outputFormat->setSpaceBetweenBlocks(' between-space ');
```

Comment:
Is this needed?

Reply:
No, it indeed isn't. I've removed it now.